tests/src/lib.rs

```diff
@@ -70,3 +70,14 @@ pub fn secret_key_to_peer_id(node_key: nomos_libp2p::ed25519::SecretKey) -> Peer
             .into(),
     )
 }
+
+#[must_use]
+pub fn secret_key_to_provider_id(
+    node_key: nomos_libp2p::ed25519::SecretKey,
+) -> nomos_sdp_core::ProviderId {
+    nomos_sdp_core::ProviderId(
+        nomos_libp2p::ed25519::Keypair::from(node_key)
+            .public()
+            .to_bytes(),
+    )
+}
```

Comment:
@bacv was this the provider id finally? Just double checking

Reply:
Yes, here is provider id definition in the spec:

> provider_id: used to sign the SDP messages and to establish secure links between validators; it is Ed25519PublicKey.

Provider in `nomos-sdp` will soon have it's own module, it'll contain the serialization and conversion between types. Will add in another PR.

[Spec](provider_id: used to sign the SDP messages and to establish secure links between validators; it is Ed25519PublicKey.)